### PR TITLE
Added doc for torch.einsum sublist format

### LINKED ISSUE #21412: Support sublist arguments for torch.einsum
## 🚀 Feature
Support the argument format `torch.einsum(op0, sublist0, op1, sublist1, ..., [sublistout])`. It would allow the users to use more than 26 tensors (i.e. the number of lower case letters) in Einstein Summation.

## Motivation

This feature is already available in [Numpy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.einsum.html).

## Pitch

Perform the operation below in PyTorch:

```python
import numpy as np

x = np.random.rand(6).reshape(2, 3)
y = np.random.rand(6).reshape(2, 3)
z = np.random.rand(6).reshape(2, 3)
w = np.random.rand(6).reshape(2, 3)
r1 = np.einsum(
            x, [0, 1], 
            y, [0, 2], 
            z, [0, 3], 
            w, [0, 4], 
            [1, 2, 3, 4]
     ) # Not currently supported by PyTorch
r2 = np.einsum('ni, nj, nk, nl -> ijkl', x, y, z, w) # Supported by PyTorch
print((r1 == r2).all())  # Prints True
```

## Additional context

I actually don't know many contexts where one really needs to call Einstein Summation on more than 26 tensors. However, for some use cases one could gain some readability if you call Einstein Summation with, say, 5 or 6 tensors.

### PR DESCRIPTION
**Pull request by heitorschueroff**
Adds documentation for the new sublist format for `torch.einsum`

closes https://github.com/pytorch/pytorch/issues/21412

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -347,6 +347,14 @@ def einsum(*args):
         run faster or consume less memory. Projects like opt_einsum (https://optimized-einsum.readthedocs.io/en/stable/)
         can optimize the formula for you.
 
+    .. note::
+
+        As of PyTorch 1.9 :func:`torch.einsum` also supports the sublist format (see examples below). In this format,
```
Comment (mruberry) [inline torch/functional.py:352]: The second and third sentences seem largely redundant.

What about:

"In this format, subscripts for each operand are specified by a "sublist" of integers in the range [0, 52). These sublists follow their operands, and an extra sublist can appear at the end of the input to specify the output's subscripts."

?

File: torch/functional.py [hunk 2]
```diff
@@ -347,6 +347,14 @@ def einsum(*args):
...
+        subscripts for an operand are specified by a sublist (list of integers in the range [0, 52)). The input in this
+        format is each input operand followed by its sublist and optionally, an extra sublist at the end for the
+        output subscripts, e.g.`torch.einsum(op1, sublist1, op2, sublist2, ..., [subslist_out])`. Python's `Ellipsis`
```
Comment (mruberry) [inline torch/functional.py:355]: Nice example

File: torch/functional.py [hunk 3]
```diff
@@ -396,6 +404,10 @@ def einsum(*args):
         >>> torch.einsum('bn,anm,bm->ba', l, A, r)
         tensor([[-0.3430, -5.2405,  0.4494],
                 [ 0.3311,  5.5201, -3.0356]])
+
+        # sublist format
+        >>> torch.einsum(A, [Ellipsis, 0], l, [1, Ellipsis], [Ellipsis]).shape
```
Comment (mruberry) [inline torch/functional.py:409]: Can you use `...` or must it actually be `Ellipsis`?

Comment (mruberry) [reply torch/functional.py:409]: Would it be more interesting to rewrite one of the above operations using the sublist format, so people could see how it looks when written "normally" and how it look when written as a sublist?

Comment (heitorschueroff) [reply torch/functional.py:409]: `...` also works and yes, I'll rewrite an example above.

File: torch/functional.py (hunk 1 above)
Comment (heitorschueroff) [inline torch/functional.py:352]: 1.9 -> 1.10

**Review by mruberry — APPROVED** [review]
Nice improvement

**Pull request closed without merging**